dashboard/app/models/section.rb

```diff
@@ -77,8 +77,8 @@ def name_safe_students
     # normalization, this is necessary to ensure that we can recognize
     # the name on the other side
     self.students.each do |student|
-      normalized = student.name.split(/\s+/).join(' ')
-      trie.add normalized
+      student.name = student.name.strip.split(/\s+/).join(' ')
```

Comment:
(note that this is definitely a discussion we can have, but not one that should prevent this PR from going out, as we're already mutating the student name below)

It might make more sense to add a field like `safe_name` rather than mutating the existing name, given that this isn't actually (as I had originally thought) going to an api of any kind, but merely a controllable rails view.